api/src/main/java/org/openmrs/module/ugandaemrreports/reports2019/SetupPNCRegister2019.java

```diff
@@ -0,0 +1,234 @@
+        dsd.addColumn("BCG", sdd.definition("BCG", getConcept("dc918618-30ab-102d-86b0-7a5022ba4115")), "onOrAfter=${startDate},onOrBefore=${endDate}", new ImmunizationDataConverter());
+        dsd.addColumn("OPV", sdd.definition("OPV", getConcept("dc918618-30ab-102d-86b0-7a5022ba4115")), "onOrAfter=${startDate},onOrBefore=${endDate}", new ImmunizationDataConverter());
+        dsd.addColumn("DPT", sdd.definition("DPT", getConcept("dc918618-30ab-102d-86b0-7a5022ba4115")), "onOrAfter=${startDate},onOrBefore=${endDate}", new ImmunizationDataConverter());
+        dsd.addColumn("Vit A", sdd.definition("Vit A", getConcept("dc918618-30ab-102d-86b0-7a5022ba4115")), "onOrAfter=${startDate},onOrBefore=${endDate}", new ImmunizationDataConverter());
+        dsd.addColumn("PCV", sdd.definition("PCV", getConcept("dc918618-30ab-102d-86b0-7a5022ba4115")), "onOrAfter=${startDate},onOrBefore=${endDate}", new ImmunizationDataConverter());
+        dsd.addColumn("Other treatment child", sdd.definition("Other treatment child", getConcept("59560ede-43e2-4e56-a47e-0f876779f0e1")), "onOrAfter=${startDate},onOrBefore=${endDate}", new ObsDataConverter());
+        dsd.addColumn("RTW/RFW", referredToOrFrom(), "onDate=${endDate}", new CalculationResultDataConverter());
+        dsd.addColumn("FSG Enrollment", sdd.definition("FSG Enrollment", getConcept("7e8e0ef3-2fda-4c76-8cf6-c7a6f1584ff2")), "onOrAfter=${startDate},onOrBefore=${endDate}", new FSGEnrollmentDataConverter());
+        dsd.addColumn("Enrolled at MBCP", sdd.definition("Enrolled at MBCP", getConcept("f0d12a70-04a3-4f7f-992b-bed4ad331908")), "onOrAfter=${startDate},onOrBefore=${endDate}", new ObsDataConverter());
+
+        return dsd;
+    }
+
+    private Concept getConcept(String uuid) {
+        return Dictionary.getConcept(uuid);
+    }
+
+    private DataDefinition villageParish(){
+        CalculationDataDefinition cdf =new CalculationDataDefinition("village+parish", new PersonAddressCalculation());
+        cdf.addParameter(new Parameter("onDate", "On Date", Date.class));
+        return cdf;
+    }
+    private DataDefinition age(Integer lower, Integer upper) {
+        CalculationDataDefinition cdf = new CalculationDataDefinition("Age-"+lower+"-"+upper+"yrs", new AgeLimitCalculation());
+        cdf.addCalculationParameter("lowerLimit", lower);
+        cdf.addCalculationParameter("upperLimit", upper);
+        cdf.addParameter(new Parameter("onDate", "On Date", Date.class));
+        return cdf;
+    }
+    private DataDefinition whoCd4Vl(String q, String a){
+        CalculationDataDefinition cd = new CalculationDataDefinition("", new WhoCd4VLCalculation());
+        cd.addParameter(new Parameter("onDate", "On Date", Date.class));
+        cd.addCalculationParameter("question", q);
+        cd.addCalculationParameter("answer", a);
+        return cd;
+    }
+
+    private DataDefinition referredToOrFrom(){
```

Comment:
@solemabrothers this should also be globally defined

Reply:
Done